packages/router-generator/tests/generator.test.ts

```diff
@@ -40,19 +40,25 @@ async function traverseDirectory(
   }
 }
 
+function shouldUseInlineConfig(folder: string) {
+  return folder !== 'config-json';
+}
+
 async function setupConfig(
   folder: string,
   inlineConfig: Partial<Omit<Config, 'routesDirectory'>> = {},
 ) {
   const { generatedRouteTree = '/routeTree.gen.ts', ...rest } = inlineConfig
   const dir = makeFolderDir(folder)
 
-  const config = await getConfig({
+  const defaultConfig = shouldUseInlineConfig(folder) === true ? {
     disableLogging: true,
     routesDirectory: dir + '/routes',
     generatedRouteTree: dir + generatedRouteTree,
     ...rest,
-  })
+  } : undefined;
```

Comment:
If this test is specifically verifying config from the filesystem, bail on passing an inline config. Use the resolved config instead.